h/static/scripts/controllers.coffee

```diff
@@ -48,7 +48,8 @@ class AppController
           annotationMapper.loadAnnotations data
         when 'delete'
           for annotation in data
-            $scope.$emit('annotationDeleted', annotation)
+            if a = threading.idTable[annotation.id]?.message
+              $scope.$emit('annotationDeleted', a)
```

Comment:
Okay just chatted with @nickstenning about this. As this is consistent with the rest of the application controller lets get this fix in as is. Could you add a test to verify the fix and to prevent regressions, then we're good to go.